docs/smtchecker.rst

```diff
@@ -19,22 +19,110 @@ SMTChecker module automatically tries to prove that the code satisfies the
 specification given by ``require/assert`` statements. That is, it considers
 ``require`` statements as assumptions and tries to prove that the conditions
 inside ``assert`` statements are always true.  If an assertion failure is
-found, a counterexample is given to the user, showing how the assertion can be
-violated.
+found, a counterexample may be given to the user, showing how the assertion can
+be violated. If no warning is given by the SMTChecker for a property,
+it means that the property is safe.
 
 The other verification targets that the SMTChecker checks at compile time are:
 
-- Arithmetic underflow and overflow (`underflow` and `overflow`).
-- Division by zero (`divByZero`).
-- Trivial conditions and unreachable code (`constantCondition`).
-- Popping an empty array (`popEmptyArray`).
-- Insufficient funds for a transfer (`balance`).
-
-The names after each target above can be used when specifying subsets of targets.
-
+- Arithmetic underflow and overflow.
+- Division by zero.
+- Trivial conditions and unreachable code.
+- Popping an empty array.
+- Insufficient funds for a transfer.
+
+The potential warnings that the SMTChecker reports are:
+
+- ``<failing  property> happens here.``. This means that the SMTChecker proved that a certain property fails. A counterexample may be given, however in complex situations it may also not show a counterexample. This result may also be a false positive in certain cases, when the SMT encoding adds abstractions for Solidity code that is either hard or impossible to express.
+- ``<failing property> might happen here``. This means that the solver could not prove either case in the given timeout. Since the result is unknown, the SMTChecker reports the potential failure for soundness. This may be solved by increasing the query timeout, but the problem might also simply be too hard.
+
+.. note::
+  The lack of warnings for a verification target represents an undisputed
+  mathematical proof of correctness. Keep in mind that these problems are
```

Comment:
This is a very strong statement. Should we add that "provided there are no bugs in the encoding or solver"?

Reply:
Well, I thought that was implicit haha
It's also only true if the specification is correct. So there are a few implicit assumptions to that statement. But I can add that, yes.